src/js/wall/FirstUser.tsx

```diff
@@ -0,0 +1,101 @@
+    /** Whether the user will be forced to reset their password on next login */
+    forceReset: boolean;
+};
+
+type FormData = {
+    username: string;
+    password: string;
+};
+
+/**
+ * A form for creating the first instance user
+ */
+export default function FirstUser() {
+    const mutation = useMutation<User, ErrorResponse, NewUser>(createFirst, {
+        onSuccess: () => {
+            window.location.reload();
+        },
+    });
+
+    const {
+        formState: { errors },
+        handleSubmit,
+        register,
+    } = useForm<FormData>();
+
+    function onSubmit(data: FormData) {
+        mutation.mutate({
+            handle: data.username,
+            password: data.password,
+            forceReset: false,
+        });
+    }
+
+    return (
+        <WallContainer>
+            <WallDialog>
+                <WallLoginContainer>
+                    <WallTitle />
+                    <WallHeader>Setup First User</WallHeader>
+                    <WallSubheader>
+                        Create an administrative user that can be used to configure your new Virtool instance.
+                    </WallSubheader>
+
+                    <form onSubmit={handleSubmit(onSubmit)}>
+                        <InputGroup>
+                            <InputLabel htmlFor="username">Username</InputLabel>
+                            <InputSimple
+                                aria-label="username"
+                                id="username"
+                                defaultValue=""
+                                {...register("username", { required: true })}
+                            />
+                            <InputError>{errors.username?.message}</InputError>
+                        </InputGroup>
+                        <InputGroup>
+                            <InputLabel htmlFor="password">Password</InputLabel>
+                            <InputSimple
+                                aria-label="password"
+                                id="password"
+                                defaultValue=""
+                                {...register("password", {
+                                    required: "Password does not meet minimum length requirement (8)",
+                                    minLength: {
+                                        value: 8,
+                                        message: "Password does not meet minimum length requirement (8)",
+                                    },
+                                })}
+                            />
+                            <InputError>{errors.password?.message}</InputError>
+                        </InputGroup>
+
+                        <WallButton type="submit" icon="user-plus" color="blue">
+                            Create User
+                        </WallButton>
+                        {mutation.isError && <InputError>{mutation.error.response?.body.message}</InputError>}
+                    </form>
```

Comment:
I think this part actually handles the error-handling for username and password. You can maybe double check if it does and remove the error-handling on password & username.

Reply:
Yup, you are right